sdk/typescript/introspector/scanner/utils.ts

```diff
@@ -164,8 +164,9 @@ type OptionalValue = {
 /**
  * Return true if the parameter is optional.
  *
- * This includes both optional value defines with `?` and value that
- * have a default value.
+ * This only includes optional value defines with `?` or `<string>|null`.
+ * If a value has a default but isn't defined with `?`, it's not considered
+ * optional in the context of GraphQL.
```

Comment:
To be technically accurate, in the context of GraphQL, it is considered optional (since it can be omitted), just not nullable.

Dagger's use of the word "optional" doesn't match GraphQL's. This may be just the english wording, but when you say "in the context of GraphQL" it implies you're using GraphQL's terminology rather than Dagger's interpretation.

Reply:
Fixed in [55606ed](https://github.com/dagger/dagger/pull/6766/commits/55606edc6b117cee7a1b579e4ea9fe9d361671ca)